File: sem2/po/PO-VirtualWorld-Python/simulator/organisms/plants/Guarana.py
from simulator.organisms.plants.Plant import *
from simulator.organisms.Organism import ORGANISM_TYPES

GUARANA_SEEDING_THRESHOLD = BASE_PLANT_SEEDING_THRESHOLD
GUARANA_STRENGTH = BASE_PLANT_STRENGTH
GUARANA_INITIATIVE = BASE_PLANT_INITIATIVE
GUARANA_STRENGTH_BONUS = 3


class Guarana(Plant):
    def __init__(self, new_world, new_location):
        super().__init__(new_world, new_location, GUARANA_STRENGTH, GUARANA_INITIATIVE, ORGANISM_TYPES["Guarana"], GUARANA_SEEDING_THRESHOLD)

    def spawn_offspring(self, new_location):
        return Guarana(self.world, new_location)

    def __str__(self):
        return "Guarana at " + str(self.location)

    def collision(self, invader):
        invader.tlog(" ate " + str(self)  + " and gained +" + str(GUARANA_STRENGTH_BONUS) + " Strength")
        invader.set_strength(invader.get_strength() + GUARANA_STRENGTH_BONUS)
        invader.update_location(self.location)
        self.kill_self()

File: sem2/po/po_proj2/src/studia/prog_ob/projekt2/world_simulator/organisms/plants/plants_subclasses/Guarana.java
package studia.prog_ob.projekt2.world_simulator.organisms.plants.plants_subclasses;

import studia.prog_ob.projekt2.world_simulator.SimLogger;
import studia.prog_ob.projekt2.world_simulator.World;
import studia.prog_ob.projekt2.world_simulator.maps.MapLocation;
import studia.prog_ob.projekt2.world_simulator.organisms.Organism;
import studia.prog_ob.projekt2.world_simulator.organisms.plants.Plant;

public class Guarana extends Plant {
    public static final int GUARANA_SEEDING_THRESHOLD = BASE_PLANT_SEEDING_THRESHOLD;
    public static final int GUARANA_STRENGTH = BASE_PLANT_STRENGTH;
    public static final int GUARANA_INITIATIVE = BASE_PLANT_INITIATIVE;
    public static final char GUARANA_SYMBOL = 'r';
    public static final int GUARANA_STRENGTH_BONUS = 3;


    public Guarana(MapLocation new_map_location, World new_world) {
        super(GUARANA_STRENGTH, GUARANA_INITIATIVE, new_map_location, GUARANA_SYMBOL, new_world, GUARANA_SEEDING_THRESHOLD);
    }

    @Override
    public Organism spawn_offspring(MapLocation location) {
        return new Guarana(location, getWorld());
    }

    @Override
    public void collision(Organism invader) {
        invader.setStrength(invader.getStrength() + GUARANA_STRENGTH_BONUS);
        SimLogger.log(invader + SimLogger.GUARANA_MESSAGE + this);
        super.collision(invader);
    }

    @Override
    public String toString() {
        return "Guarana " + getMap_location() + " ";
    }
}
